engine/apps/auth_token/auth.py

```diff
@@ -359,6 +358,12 @@ def authenticate(self, request):
         return self.authenticate_credentials(organization, auth)
 
     def get_organization(self, request):
+        grafana_url = request.headers.get(X_GRAFANA_URL)
+        if grafana_url:
+            organization = Organization.objects.filter(grafana_url=grafana_url).first()
+            if organization:
+                return organization
+
```

Comment:
We need a way to identify the Grafana URL to hit its API when validating a service account token (note Grafana URL is already a param in [Terraform](https://registry.terraform.io/providers/grafana/grafana/latest/docs#url-1))